src/mrpro/operators/models/_TransientInversionRecovery.py

```diff
@@ -0,0 +1,190 @@
+    start of the data acquisition, the time points of the inversion pulses and the gap between inversion pulses and the
+    first data acquisition, the sequence is fully described. This allows for the calculation of a signal from any
+    arbitrary subset of the data acquisition.
+
+
+    More information on this signal model can be found in:
+    Becker, K. M., Schulz-Menger, J., Schaeffter, T. & Kolbitsch, C.
+    Simultaneous high-resolution cardiac T1 mapping and cine imaging using model-based iterative image reconstruction.
+    Magn. Reson. Med. 81, 1080-1091 (2019). http://doi.wiley.com/10.1002/mrm.27474
+    """
+
+    def __init__(
+        self,
+        signal_time_points: torch.Tensor,
+        tr: float | torch.Tensor,
+        inversion_time_points: torch.Tensor,
+        time_inversion_adc: float | torch.Tensor,
+        first_adc_time_point: float | torch.Tensor | None = None,
+    ):
+        """Initialize continuous acquisition with inversion pulses.
+
+        Parameters
+        ----------
+        signal_time_points
+            time stamp of each acquisition
+        tr
+            repetition time
+        inversion_time_points
+            time stamp of each inversion
+        time_inversion_adc
+            time between inversion pulse and start of data acquisition
+        first_adc_time_point
+            time stamp of first acquisition
+        """
+        super().__init__()
+        tr = torch.Tensor(tr)
+        time_inversion_adc = torch.Tensor(time_inversion_adc)
+
+        self.signal_time_points = torch.nn.Parameter(signal_time_points, requires_grad=signal_time_points.requires_grad)
+        self.tr = torch.nn.Parameter(tr, requires_grad=tr.requires_grad)
+        self.time_inversion_adc = torch.nn.Parameter(time_inversion_adc, requires_grad=time_inversion_adc.requires_grad)
+
+        self.first_adc_time_point = torch.Tensor(first_adc_time_point) if first_adc_time_point is not None else None
+        self.index_before_first_inversion = torch.where(signal_time_points < inversion_time_points[0])[0]
+
+        if len(self.index_before_first_inversion) > 0 and self.first_adc_time_point is None:
+            raise ValueError(
+                'If data has been acquired before the first inversion pulse,',
+                'the start of the acquisitions first_adc_time_point has to be defined.',
+            )
+
+        # Verify that no acquisition occurred before start of acquisition
+        if self.first_adc_time_point is not None and any(signal_time_points < first_adc_time_point):
+            raise ValueError('Acquisitions detected before start of acquisition.')
+
+        # Add final "virtual" inversion pulse to allow for easier looping later
+        inversion_time_points = torch.concat((inversion_time_points, (signal_time_points.max() + 1).reshape(1)), dim=0)
+        self.inversion_time_points = torch.nn.Parameter(
```

Comment:
is the function a) differentiable wrt to the inversion time points? If not, this should not be a Parameter but a buffer.
b) can you later on change the inversion time points? If not, it should be _inversion_time_points with an underscore, right?

Sorry, I did not remember the content of this function when you asked the question earlier..

Reply:
I don't think that the signal is differentiable wrt to any of the timings because they are dependent on each other. 

If I simply do `self.register_buffer('_inversion_time_points', inversion_time_points)` then at least in my VSCode `self._inversion_time_points` is of type Any and I lose all my type hinting. How is `register_buffer` usually used?